modules/FiniteSetsExt.tla

```diff
@@ -69,16 +69,24 @@ Min(S) == CHOOSE x \in S : \A y \in S : x =< y
 (* If Sets is a set of (non-empty) sets then Choice(Sets) is the set of    *)
 (* all choice functions over Sets, that is, functions that associate some  *)
 (* with every set in Sets.                                                 *) 
+(*                                                                         *)
+(* Example:                                                                *)
+(*          Choice({{1,2}, {3,4}, {5}}) =                                  *)
+(*                           ({5} :> 5 @@ {1, 2} :> 1 @@ {3, 4} :> 3),     *)
+(*                           ({5} :> 5 @@ {1, 2} :> 1 @@ {3, 4} :> 4),     *)
+(*                           ({5} :> 5 @@ {1, 2} :> 2 @@ {3, 4} :> 3),     *)
+(*                           ({5} :> 5 @@ {1, 2} :> 2 @@ {3, 4} :> 4) }    *)
 (***************************************************************************) 
 Choice(Sets) == { f \in [Sets -> UNION Sets] : \A S \in Sets : f[S] \in S }
 
 -----------------------------------------------------------------------------
 
-Range(f) == {f[x] : x \in DOMAIN f}
-
 (***************************************************************************) 
 (* Compute all sets that contain one element from each of the input sets:  *)
-(* Shuffle({{1,2}, {3,4}, {5}}) = {{1,3,5}, {1,4,5}, {2,3,5}, {2,4,5}}     *)
+(*                                                                         *)
+(* Example:                                                                *)
+(*          Shuffle({{1,2}, {3,4}, {5}}) =                                 *)
+(*                         {{1,3,5}, {1,4,5}, {2,3,5}, {2,4,5}}            *)
 (***************************************************************************) 
 Shuffle(Sets) == { Range(f) : f \in Choice(Sets) }
```

Comment:
The name `Shuffle` is IMO not a good fit because it indicates an order where there is none.  Does the alternative definition, built out of the `kSubset` operator, get us any closer to a less ambiguous name for this family of sets? Perhaps, something along the lines of "cover" such as `kSubsetCover` (for some def of cover)?

```tla
kSubsetCover(S) == { subsets \in SUBSET (UNION S) : 
     /\ Cardinality(subsets) = Cardinality(S) 
     \* Cover every set in S.
     /\ \A s \in S: \E e \in subsets: e \in s
 }
```

Reply:
I don't know if I understand this correctly. The suggestion is to leave `Choice` as is, but only replace `Shuffle` with `kSubsetCover`? Also, where does `kSubset` come in this definition of `kSubsetCover`? 

Reply:
I agree that "Shuffle" is not a good name, but "(subset) cover" has an established meaning and is different. Perhaps something like "Draws"? Also, given that this is going to end up in a "standard" module, "Choice" should perhaps be renamed to "ChoiceFunctions" or something like that?

Reply:
I like `Draws` better than `Shuffle` or `kSubsetCover` but it carries very little meaning (how are elements drawn?).  

Regardless of its name, a Java module overwrite should be easy to be built from the overwrite of [`kSubset`](https://github.com/tlaplus/CommunityModules/blob/2ebcbd910bcf5a2cc7f62cdb3a11c9e2f2d15641/modules/tlc2/overrides/FiniteSetsExt.java#L83-L103).

Reply:
The example should, perhaps, cover the case where the nested/input sets are *not* disjoint?

Reply:
There doesn't seem to be an established mathematical term for this operator.  I suggest we just go with `DifferentOneOfEach` as [suggested online](https://twitter.com/pressron/status/1320786781839724546) and rename it if we find a better term.

Reply:
Shouldn't `Cardinality(subsets) = Cardinality(S)` be changed to `Cardinality(subsets) <= Cardinality(S)`? Only then this new definition is equivalent to the old one:
```
DifferentOneOfEach(S) == { subsets \in SUBSET (UNION S) : 
     /\ Cardinality(subsets) <= Cardinality(S) 
     \* Cover every set in S.
     /\ \A s \in S: \E e \in subsets: e \in s
 }
```